src/javaRosa/util.js

```diff
@@ -47,16 +47,21 @@ define([
         return getDefaultItextRoot(mug) + "-" + property;
     };
 
-    var looksLikeMarkdown = function(val) {
+    var looksLikeMarkdown = function(val, supportTables) {
         /* Regex checks (in order):
          * ordered lists
          * unordered lists
          * strikethrough
          * headings
          * italics/bold/bold italics
          * links
+         * tables (regex written after ||)
          */
-        return /^\d+[\.\)] |^\* |~~.+~~|# |\*{1,3}\S.*\*{1,3}|\[.+\]\(\S+\)/m.test(val);
+        if (supportTables) {
+            return /^\d+[\.\)] |^\* |~~.+~~|# |\*{1,3}\S.*\*{1,3}|\[.+\]\(\S+\)/m.test(val) || /^(\|[^\n]+\|\r?\n)((?:\|:?[-]+:?)+\|)(\n(?:\|[^\n]+\|\r?\n?)*)?$/m.test(val);
+        } else {
+            return /^\d+[\.\)] |^\* |~~.+~~|# |\*{1,3}\S.*\*{1,3}|\[.+\]\(\S+\)/m.test(val);
+        }
```

Comment:
The duplication could be removed here by using variables.
```suggestion
        var hasMarkdown = /^\d+[\.\)] |^\* |~~.+~~|# |\*{1,3}\S.*\*{1,3}|\[.+\]\(\S+\)/m.test(val),
            hasTable = false;
        if (supportTables) {
            hasTable = /^(\|[^\n]+\|\r?\n)((?:\|:?[-]+:?)+\|)(\n(?:\|[^\n]+\|\r?\n?)*)?$/m.test(val);
        }
        return hasMarkdown || hasTable;
```

Reply:
Sweet 😄